packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts

```diff
@@ -888,6 +888,12 @@ export class AppSyncModelJavaVisitor<
       read: 'ModelOperation.READ',
       update: 'ModelOperation.UPDATE',
       delete: 'ModelOperation.DELETE',
+      //granular read
+      get: 'ModelOperation.GET',
+      list: 'ModelOperation.LIST',
+      sync: 'ModelOperation.SYNC',
+      listen: 'ModelOperation.LISTEN',
+      search: 'ModelOperation.SEARCH',
```

Comment:
Are we changing the behavior for only `appsync-java-visitor` or all other visitors as well? I assume this `generateAuthRules` method is common to all visitors? If that's the case, can we move this to a common visitor? Thanks

Reply:
This is for java visitor only. It maps the input operation to the library-defined enum value. If not added here, the code in java will be generated as empty string.
For the other code generators, these ops will be simply parsed and generated in the final code and do not need additional mapping. You can see this in the snapshot changes for other visitors

Reply:
- Swift: https://github.com/aws-amplify/amplify-codegen/pull/537/files#diff-b728b3bfd6e02d0683386efa2a8a74b95cc6f624d5f450db27764fe7edfac671R53-R55
- JS: https://github.com/aws-amplify/amplify-codegen/pull/537/files#diff-cc6588d1d5a98c0aba946398191679c0e2c1dbc8cc27d0c8b61e1a6d10cf34eeR2781-R2797
- Flutter: https://github.com/aws-amplify/amplify-codegen/pull/537/files#diff-38d85adcb8cc38618dae4676dc8ef3ae8c13ecf2fc73bbb00ce96519e7cbe2c6R3109-R3115